Bump sea-builder's listr2 to v11 (and the @types/node it requires)

## Background

`packages/sea-builder/package.json` pins `listr2` at `^9.0.4` (resolving
to `9.0.5`). npm `latest` is `11.0.0`, released 2026-07-21. This issue
carries the `listr2` half of #125, split out because the empirical
picture has changed: the `listr2` bump no longer needs #120.

Split out of #125.

### What v10 and v11 actually change

Enumerated from the upstream release notes of **every** release between
`9.0.5` and `11.0.0` (`listr2@10.0.0`, `10.1.0`, `10.1.1`, `10.1.2`, `10.2.0`,
`10.2.1`, `10.2.2`, `11.0.0` plus the `10.3.0-beta.1` / `11.0.0-beta.*`
series), then matched against this package's actual usage.

**No release in that range deprecates anything.** Upstream ships no
deprecation notices across v10 and v11 — every item below is either a
hard breaking change or an internal/behavioural change, so there is no
"still works but is now legacy" API left behind in this package.

| Upstream change                                                                                                  | Release  | Kind        | Impact on `sea-builder`                                                                                                                                     |
| ---------------------------------------------------------------------------------------------------------------- | -------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| Node.js 20 support dropped; `engines.node` becomes `>=22.13.0`                                                   | `10.0.0` | breaking    | None. This package's floor is `^22.23.2` (`packages/sea-builder/package.json`), `.node-version` is `22.23.2`, and CI's matrix is `22.x`/`24.x`/`26.x`.        |
| Bundle mechanism reworked                                                                                        | `10.1.0` | internal    | None. Consumed only through the package's public entry.                                                                                                      |
| `colorette` replaced with `node:util`'s `styleText`                                                              | `10.2.0` | internal    | None at runtime — but this is the origin of the one type-level blocker below.                                                                                 |
| `collectErrors` no longer accepts `'minimal'` / `'full'`; it is a boolean. `ListrError.ctx`, the `cloneObject` export and the `rfdc` dependency are removed | `11.0.0` | breaking    | None. The only option object this package passes is `{ concurrent: true }` (`packages/sea-builder/src/listr2/createCacheTasks.mts:13`); `createBuildTasks` passes none at all (`.../createBuildTasks.mts:32`). Nothing imports `cloneObject`, `ListrError`, or `ctx`. |
| `Listr.errors` becomes `ListrError[] \| null` (`null` under the default `collectErrors`)                          | `11.0.0` | breaking    | None. `errors` is never read. `src/builder.mts:21-26` handles failure by catching the `run()` rejection; `src/cache.mts:12` lets it reject.                    |
| On interruption, tasks without a rollback become `CANCELLED` instead of `FAILED`; exit waits for in-flight rollbacks. Custom renderers must handle `ListrTaskState.CANCELLED` | `11.0.0` | breaking    | Behaviour-only. No custom renderer and no `ListrTaskState` import exist; the change is visible only as different Ctrl-C output.                                |
| `eventemitter3` replaced with Node's built-in `EventEmitter`                                                      | `11.0.0` | internal    | None. No event listeners are attached.                                                                                                                        |
| Task list re-rendered with partial updates via `log-update` v8                                                   | `11.0.0` | cosmetic    | None beyond redraw behaviour.                                                                                                                                 |
| Package becomes ESM-only: `exports` maps both `require` and `import` to `dist/index.mjs`, types to `dist/index.d.mts` | `11.0.0` | breaking    | None. `sea-builder` is `"type": "module"`, emits `.mjs`, and keeps `listr2` external rather than bundling it (`dist/builder.mjs` is 4.74 kB after `vite build`). |

The complete `listr2` surface this package touches is three things —
`new Listr(...)` (`src/listr2/createBuildTasks.mts:32`,
`src/listr2/createCacheTasks.mts:13`), `.run()` (`src/builder.mts:19`,
`src/cache.mts:12`, `src/tasks/createCacheTask.mts:14`), and `Listr` as a
return type (`src/listr2/createBuildTasks.mts:28`,
`src/listr2/createCacheTasks.mts:12`). Task objects come from this
package's own structural `Task` interface
(`src/tasks/createTaskFactory.mts:10-16`), not from `ListrTask`, so no
upstream type rename can reach them. **The migration therefore needs no
source change at all.**

### The one real blocker: a type this `@types/node` major does not export

`listr2@11`'s `dist/index.d.mts` opens with
`import { InspectColor } from "node:util"` — a consequence of the
`10.2.0` switch to `styleText`. `@types/node@24` does not export that
name. Inspecting the published tarballs of `22.19.1`, `24.6.2`,
`24.13.3`, `25.0.0`, `25.9.3`, `26.0.0` and `26.2.0` places its first
appearance in **`@types/node@25.0.0`**. Node.js 25 is outside this
repository's `engines.node`, so `26.x` is the actionable major.

With `listr2` at `11.0.0` and `@types/node` untouched, `pnpm run test:ts`
fails with exactly one error and no cascade:

```text
listr2/dist/index.d.mts(1,10): error TS2305: Module '"node:util"' has no exported member 'InspectColor'.
```

### Why this is no longer blocked by #120

#125's 2026-08-13 re-verification recorded that `listr2@^11.0.0` plus
`@types/node@^26.2.0` in `sea-builder` alone still failed `test:ts` with
duplicate-identifier errors. **That combination now passes.** Re-run
against the current tree in disposable scratch copies (working tree
untouched throughout), one clean copy per configuration:

| Configuration (exact pins, no `^`)                                                        | `build` | `test:ts` | `test:vitest` | `lint` |
| ------------------------------------------------------------------------------------------- | ------- | --------- | ------------- | ------ |
| baseline (`listr2@9.0.5`, `@types/node@^24.6.2`)                                           | pass    | pass      | pass          | pass   |
| `listr2@11.0.0` only                                                                       | pass    | **fail**  | pass          | pass   |
| `listr2@11.0.0` + `@types/node@26.0.0` in `sea-builder` only                                | pass    | pass      | pass          | pass   |
| `listr2@11.0.0` + `@types/node@26.2.0` in `sea-builder` only                                | pass    | pass      | pass          | pass   |
| `listr2@11.0.0` + `@types/node@26.0.0` in **both** `sea-builder` and `vite-lib-config`      | pass    | **fail**  | pass          | pass   |

Two conclusions follow. First, the `sea-builder`-only bump is actionable
today and does not wait on #120. Second, #125's own remaining
deliverable — aligning every package on one `@types/node` major — is
still genuinely blocked: the last row fails inside **`example-lib`**, a
package that declares no `@types/node` of its own, with `TS2300`
duplicate-identifier and `TS2428` errors raised against the
`@types/node@24.13.3` copy that stays in the graph transitively.

The mechanism behind the change since 2026-08-13 was not diagnosed. The
only workspace movement in that window was the pnpm bump to `11.21.0`
and the `.node-version` bump, so re-confirm the third/fourth row before
relying on it rather than treating this table as settled.

One trap worth carrying forward: running these configurations
incrementally in a single scratch copy left **three** `@types/node`
majors in `pnpm-lock.yaml` and reproduced the duplicate-identifier
failure spuriously. Verify each configuration from a fresh copy, and
check the resolved set with
`grep -oE "@types/node@[0-9]+\.[0-9]+\.[0-9]+" pnpm-lock.yaml | sort -u`.
Also run `pnpm run build` before `pnpm run test:ts` in a fresh copy —
`vite-lib-config`'s built output has to exist first, or unrelated
`TS2307` errors appear in `vite.config.mts` / `vitest.config.mts`.

### Alternatives that were verified and rejected

Two hatches avoid touching `@types/node` at all, and both were confirmed
to pass `build` / `test:ts` / `test:vitest` / `lint` in clean scratch
copies. Neither is what this issue asks for; they are recorded so the
implementer does not re-derive them:

- `"skipLibCheck": true` in `packages/sea-builder/tsconfig.json` — one
  line, but it silences declaration checking for every dependency in the
  package, and this workspace sets `skipLibCheck` nowhere.
- a local `declare module 'node:util' { export type InspectColor = string }`
  shim under `packages/sea-builder/src/` — narrow, but it is a
  hand-maintained fiction that would have to be retired later. (A
  faithful variant aliasing `@types/node@24`'s own
  `BackgroundColors | ForegroundColors | Modifiers` does **not** compile:
  those names are not in scope inside a module augmentation, `TS2304`.)

The maintainer chose the real-types route: bump `@types/node`, leave no
hatch behind.

### Runtime behaviour was checked, not assumed

The built CLI was run against `listr2@11.0.0`:
`node packages/sea-builder/dist/cache.mjs linux-x64 win32-x64` rendered
both tasks concurrently through the nested-`Listr` path
(`createCacheTask` → `createListrCacheTasks`) and reported per-task
success and failure correctly.

## Proposed change

1. In `packages/sea-builder/package.json`, bump `dependencies.listr2`
   from `^9.0.4` to `^11.0.0`.
2. In the same file, bump `devDependencies.@types/node` from `^24.6.2`
   to `^26.2.0`.
3. Regenerate `pnpm-lock.yaml`.
4. Change nothing under `packages/sea-builder/src/`. The enumeration
   above establishes that no v10/v11 breaking change reaches this
   package's code, and the two hatches listed under "Alternatives" are
   deliberately not part of this change.
5. Leave `packages/vite-lib-config`'s `@types/node` at `^24.6.2`.
   Bumping both together is verified to break `example-lib`'s typecheck;
   workspace-wide alignment stays #125's deliverable.
6. Leave `pnpm-workspace.yaml`'s
   `packageExtensions.undici-types.dependencies.@types/node` entry at
   `^24.6.2`. The verified-passing configuration left it untouched;
   revisit it only if a check actually fails.

### Version-floor note

`@types/node@26.0.0` was verified to pass just as `26.2.0` does, so a
stricter reading of the repository's oldest-that-works policy would
target `^26.0.0`. The maintainer selected `^26.2.0`; both data points are
recorded here so the floor can be lowered later without re-testing.
`listr2` has no such choice — `11.0.0` is the only stable v11 release.

## Acceptance criteria

- `packages/sea-builder/package.json` declares `listr2` as `^11.0.0` and
  `@types/node` as `^26.2.0`.
- No other workspace package's `@types/node` declaration changes, and no
  file under `packages/sea-builder/src/` changes — in particular no
  `skipLibCheck` entry and no `node:util` augmentation shim is added.
- `grep -oE "@types/node@[0-9]+" pnpm-lock.yaml | sort -u` reports
  exactly `@types/node@24` and `@types/node@26`, with no third major.
- `pnpm install --frozen-lockfile` succeeds against the regenerated
  lockfile.
- `pnpm run build` passes, and `pnpm run test:ts` passes when run after
  it.
- `pnpm run test` passes.
- `pnpm run lint:fix` then `pnpm run lint` both pass.

---

_Autopilot suitability: 5 / 5 -- higher is more autopilot-suitable;
below the configured floor is human-oriented._

<!-- builder-config-autopilot-suitability: 5 -->

_Effort: S -- author-estimated size; a soft autopilot selection
tie-breaker only._

<!-- builder-config-effort: S -->

--- a/packages/sea-builder/package.json
+++ b/packages/sea-builder/package.json
@@ -41,14 +41,14 @@
     "@types/semver": "^7.8.0",
     "all-node-versions": "^13.0.1",
     "execa": "^10.0.0",
-    "listr2": "^9.0.4",
+    "listr2": "^11.0.0",
     "node-releases": "^2.0.53",
     "semver": "^7.8.1"
   },
   "devDependencies": {
     "@kurone-kito/typescript-config": "workspace:^",
     "@kurone-kito/vite-lib-config": "workspace:^",
-    "@types/node": "^24.6.2",
+    "@types/node": "^26.2.0",
     "@vitest/coverage-v8": "^4.1.0",
     "cpy-cli": "^7.0.0",
     "rimraf": "^6.0.1",

